pallets/connectors/src/lib.rs

```diff
@@ -329,22 +387,30 @@ pub mod pallet {
 				BadOrigin
 			);
 
-			// Now add the destination address as a TrancheInvestor of the given tranche
-			T::Permission::add(
+			// Now add the destination address as a TrancheInvestor of the given tranche if
+			// not already one. This check is necessary shall a user have called `update_member`
+			// already but the call has failed on the EVM side and needs to be retried.
+			if !T::Permission::has(
 				PermissionScope::Pool(pool_id),
-				address.into_account_truncating(),
+				domain_address.into_account_truncating(),
 				Role::PoolRole(PoolRole::TrancheInvestor(tranche_id, valid_until)),
-			)?;
+			) {
+				T::Permission::add(
+					PermissionScope::Pool(pool_id),
+					domain_address.into_account_truncating(),
+					Role::PoolRole(PoolRole::TrancheInvestor(tranche_id, valid_until)),
+				)?;
+			}
```

Comment:
An idea for this. Since both parameters used are the same on `has()` and `add()`, and seems it can be a recurrent pattern, why not add a new method `add_idempotent()` (or similar) to the trait `Permissions` that does exactly this? The method could have a default implementation using only `has()` and `add()` so there is no implementation on the permission pallet.

Reply:
I actually wanted to suggest that we don't fail with `RoleAlreadyGiven` in the Permissions pallet with a permission was already set since it's not _really_ an error in my view. Wdyt?

Reply:
I like too, but not sure if there are other places that expect not to add a `Role` if it's already been added. But it makes sense to have this idempotent 👍🏻

Reply:
Created a separate issue for that: https://github.com/centrifuge/centrifuge-chain/issues/1204